examples/lighting-app/qpg/src/AppTask.cpp

```diff
@@ -118,16 +118,21 @@ void OnTriggerIdentifyEffect(Identify * identify)
     case EMBER_ZCL_IDENTIFY_EFFECT_IDENTIFIER_BLINK:
     case EMBER_ZCL_IDENTIFY_EFFECT_IDENTIFIER_BREATHE:
     case EMBER_ZCL_IDENTIFY_EFFECT_IDENTIFIER_OKAY:
-        (void) chip::DeviceLayer::SystemLayer().StartTimer(chip::System::Clock::Seconds16(5), OnTriggerIdentifyEffectCompleted,
-                                                           identify);
+        SystemLayer().ScheduleLambda([identify] {
```

Comment:
The documentation for ScheduleLambda says:
```
     *   Schedules a lambda object to be run as soon as possible in the CHIP context. This function is not thread-safe,
     *   it must be called with in the CHIP context
```

so how is this helping anything?

Reply:
Or is that documentation wrong?  In that case we should adjust the documentation....

Reply:
From what I saw, similar approaches exist in other example apps. In our tests, there was no issue using it - but indeed that's no guarantee. Also, there is a conflicting comment in NetworkCommissioning.h (line 247):

`         * OnFinished() must be called in a thread-safe manner with CHIP stack. (e.g. using ScheduleWork or ScheduleLambda)`

so indeed, maybe the thread safety aspect needs revisiting.

Reply:
`ScheduleWork` on the `SystemLayer` is absolutely not threadsafe.

`ScheduleWork` on the `PlatformManager` is threadsafe.

Reply:
By looking at ScheduleLambda and PlatformEventing::ScheduleLambdaBridge, I would say they are thread safe due to posting the event to PlatformMgr queue - similar mechanism to ScheduleWork - PlatformMgr version.

Reply:
Right, that's what I said above.  The question is whether any of the code before that does something weird (e.g. in its constructors, etc).

Reply:
[ScheduleLambda_Audit.pdf](https://github.com/project-chip/connectedhomeip/files/11454697/ScheduleLambda_Audit.pdf)
Adding audit for ScheduleLambda thread safety.